sklearn/metrics/setup.py

```diff
@@ -26,6 +26,12 @@ def configuration(parent_package="", top_path=None):
         libraries=libraries,
     )
 
+    config.add_extension(
+        "_pairwise_distances_reduction",
+        sources=["_pairwise_distances_reduction.pyx"],
+        libraries=libraries,
+    )
```

Comment:
As we `cimport numpy`, I'm surprised that we do not need the usual `include_dirs=[np.get_include(), os.path.join(np.get_include(), "numpy")]`.

Reply:
I think this unexpectedly works because other extensions defined in this setup for `metrics` do specify numpy's include folders (IIRC/IIUC, internally Cython performs a first pass on all `pxd` headers prior to compiling sources).

For safety because `_pairwise_distances_reduction` needs the numpy's include folders, I will specify `include_dirs` here.